enterprise/kernel/src/test/java/org/neo4j/kernel/impl/store/format/highlimit/NodeRecordFormatTest.java

```diff
@@ -123,6 +123,32 @@ public void useVariableLengthFormatWhenPropertyReferenceTooBig() throws IOExcept
         verifySameReferences( source, target);
     }
 
+    @Test
+    public void useVariableLengthFormatWhenRecordSizeIsTooSmall() throws IOException
+    {
+        NodeRecord source = new NodeRecord( 1 );
+        NodeRecord target = new NodeRecord( 1 );
+        source.initialize( true, randomFixedReference(), true, randomFixedReference(), 0L );
+
+        writeReadRecord( source, target, NodeRecordFormat.FIXED_FORMAT_RECORD_SIZE - 1 );
+
+        assertFalse( "Record should use variable length reference if format record is too small.", target.isUseFixedReferences() );
+        verifySameReferences( source, target);
+    }
```

Comment:
Lets take for example `HighLimitWithSmallRecords` record format in this case record size is smaller then required number of bytes, so we will not gonna be able to fit in a single unit